test/cors.test.js

```diff
@@ -942,3 +942,39 @@ test('Should support wildcard config /2', t => {
     t.equal(res.headers['access-control-allow-origin'], '*')
   })
 })
+
+test('should support embedded cors registration with route params', t => {
+  t.plan(3)
+
+  const fastify = Fastify()
+
+  const custom = async (instance, opts) => {
+    instance.register(cors, instance => async (req, cb) => {
+      const corsOptions = {
+        origin: ['example.com']
+      }
+      cb(null, corsOptions)
+    })
```

Comment:
```suggestion
    instance.register(cors, {
      origin: ['example.com']
    })
```

Reply:
The reason I am registering cors like this is so that I can use a custom origin depending on the route params. is there another way to implement that ?